TrackletAlgorithm/MatchEngine.h

```diff
@@ -104,18 +67,225 @@ constexpr unsigned int kZAdjustment = 8;
 
 /////////////////////////////
 // -- MATCH ENGINE FUNCTIONS
-void readTable(bool table[LSIZE]);
+inline void readTable(ap_uint<1> table[LSIZE]) {
+	ap_uint<1> tmp[LSIZE]=
+	#if LAYER == 1
+#include "../emData/ME/tables/METable_L1.tab"
+	#elif LAYER == 2
+#include "../emData/ME/tables/METable_L2.tab"
+	#elif LAYER == 3
+#include "../emData/ME/tables/METable_L3.tab"
+	#elif LAYER == 4
+#include "../emData/ME/tables/METable_L4.tab"
+	#elif LAYER == 5
+#include "../emData/ME/tables/METable_L5.tab"
+	#elif LAYER == 6
+#include "../emData/ME/tables/METable_L6.tab"
+	#else
+{};
```

Comment:
@aperloff It would make more sense to me to make `LAYER` a template parameter of this function, and to not define the values of `LAYER` and `PROJECTIONTYPE` in this file. For all the processing modules, we want the code outside the \*Top.\* files to be as generic as possible.

Reply:
Okay, I have changed this (and a lot more). Now the only user configurable parameter is the layer/disk number. Everything else is computed from this. This function is now templated and specialized as needed.